### ISSUE TITLE
DataCompressorV3 addresses not returned from AddressProviderV3

### ISSUE THREAD

**Issue by scottnoidea** [Gearbox-protocol/core-v3#293]
Version number 2 is hard coded in the getDataCompressor() function, preventing the return of the DataCompressorV3 address from the AddressProviderV3 functions.

```
/// @notice Data compressor contract address
    function getDataCompressor() external view returns (address) {
        return getAddressOrRevert(AP_DATA_COMPRESSOR, 2);
    }
```

The DataCompressorV3 addresses have been set with version equal to `300`:

- Ethereum - event 310 here: https://etherscan.io/tx/0xa8bae0b3f4fdf4439c9fe432af0ea6d099264c3fa37b59007af84cb8b9deffbc#eventlog
- Arbitrum - event 18 here: https://arbiscan.io/tx/0x56fe70ce2d5a989f8550d656983dcf0646c327d49f9e2587f06c48630cf84ed0
- Optimism - event 5 here: https://optimistic.etherscan.io/tx/0x8db9acf325a4a0effa8b7b81766406f7721ec85dfb51b751d8e30d6bc4c00c1a#eventlog

Note, I believe this is also preventing the DataCompressorV3 addresses from being published in the docs due to autogeneration

**Comment by lekhovitsky** [comment]
`getDataCompressor` and similar functions in `AddressProviderV3` are there for backward compatibility since the contract has been used for both V2 and V3. To properly query V3 addresses, `getAddressOrRevert` shall be used, like so:

```bash
cast call $ADDRESS_PROVIDER \
    "getAddressOrRevert(bytes32,uint256) returns (address)" \
    $(cast format-bytes32-string DATA_COMPRESSOR) 300 \
    --rpc-url $RPC_URL
```

**Comment by scottnoidea** [comment]
Thanks for the reply.

I got here because I couldn't find the DataCompressorV3 addresses.  I figured the `getDataCompressor` method should do that for me, but I guess I was mistaken.  

The addresses aren't in the docs because the docs autogen from the `AddressProviderV3` contract.  No-one in the discord knew the addresses either. I found them by going through the `SetAddress` events on each `AddressProviderV3` until I found them.

The DataCompressor is well covered in the docs, Such a useful contract shouldn't be so hard to find.  I can see your suggested method in the docs now I know about it, but it wasn't obvious this was the way.

I was hoping to make this easier for those following me who are looking for the addresses. Thanks for pointing me towards the right method.
